### PR TITLE
perf: Throttle prune job

### PR DESCRIPTION
**Pull request by adityapk00**
## Motivation

The prune job is running too fast, causing spikes in merge latency. This PR throttles the job

## Change Summary

  // We prune at the rate of 1000 fids per second. If we are running ahead of schedule, we sleep to catch up
  // We do this so that:
  // 1. Each fid is pruned at the same time at every hub, reducing thrash
  // 2. Don't overload the DB or Sync Trie, causing spikes in latency

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [X] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [X] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [X] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [ ] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [X] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

---

## PR-Codex overview
The focus of this PR is to improve the performance of the prune job in the `@farcaster/hubble` module.

### Detailed summary
- Throttled the prune job to avoid overloading the system
- Removed unnecessary parameter in `PruneMessagesJobScheduler` constructor
- Added logic to sleep and catch up if pruning is ahead of schedule

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/hubble/src/storage/jobs/pruneMessagesJob.ts [hunk 1]
```diff
@@ -82,13 +84,21 @@ export class PruneMessagesJobScheduler {
...
-      // Sleep for a bit avoid overloading the Merkle Trie
-      const syncTrieQSize = this._getSyncTrieQSizeFn();
-      if (syncTrieQSize > 10_000) {
-        log.info({ syncTrieQSize }, "sync trie Q is large, sleeping for 30s");
-        await sleep(30_000);
+        // We prune at the rate of 1000 fids per second. If we are running ahead of schedule, we sleep to catch up
+        // We do this so that:
+        // 1. Each fid is pruned at the same time at every hub, reducing thrash
+        // 2. Don't overload the DB or Sync Trie, causing spikes in latency
+        if (fid % 100 === 0) {
```
Comment (Wazzymandias) [inline apps/hubble/src/storage/jobs/pruneMessagesJob.ts:92]: question: This _may_ sleep every 100 fids - is that intended?

Comment (adityapk00) [reply apps/hubble/src/storage/jobs/pruneMessagesJob.ts:92]: Yes, it should pause every 100 fids to throttle

**Review by Wazzymandias — APPROVED** [review]

File: apps/hubble/src/storage/jobs/pruneMessagesJob.ts [hunk 2]
```diff
@@ -8,6 +8,10 @@ import { sleep } from "../../utils/crypto.js";
 
 export const DEFAULT_PRUNE_MESSAGES_JOB_CRON = "0 */2 * * *"; // Every two hours
 
+// How much time to allocate to pruning each fid.
+// 1000 fids per second = 1 fid per ms. 500k fids will take under 10 minutes
+const TIME_SCHEDULED_PER_FID_MS = 1;
```
Comment (sanjayprabhu) [inline apps/hubble/src/storage/jobs/pruneMessagesJob.ts:13]: Do you know what's the current rate? Wonder if even this is too aggressive, especially for lower spec'd machines. But we can tune as we see the impact

**Comment by Wazzymandias** [comment]
It could be nice to have to collect statsd metric on # processed FIDs vs total # fids to make sure we don't fall behind on registration spikes. But that probably isn't necessary

**Review by sanjayprabhu — APPROVED** [review]

File: apps/hubble/src/storage/jobs/pruneMessagesJob.ts (hunk 2 above)
Comment (adityapk00) [reply apps/hubble/src/storage/jobs/pruneMessagesJob.ts:13]: Currently (on the prod hubs), the whole thing takes ~6 minutes. This PR will spread it out over ~9 minutes, with breaks every 100 fids

**Pull request merged**